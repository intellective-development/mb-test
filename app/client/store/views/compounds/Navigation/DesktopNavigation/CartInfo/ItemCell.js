// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import bindClassNames from 'shared/utils/bind_classnames';
import I18n from 'store/localization';
import formatCurrency from 'shared/utils/format_currency';
import { dispatchAction } from 'shared/dispatcher';

import { cart_item_helpers, cart_item_actions } from 'store/business/cart_item';
import { product_grouping_helpers } from 'store/business/product_grouping';

import { MBLink, MBText } from '../../../../elements';
import styles from './CartDropdown.scss';

const cn = bindClassNames(styles);

type ItemCellProps = {item: Object, removeCartItem: typeof cart_item_actions.removeCartItem, show_remove_button?: boolean};
const ItemCell = ({item, removeCartItem, show_remove_button = true}: ItemCellProps) => {
  if (!item) return null;

  return (
    <li className={styles.cmCartDropdown_ItemRow_Animation}>
      <div className={styles.cmCartDropdown_ItemRow}>
        <MBLink.View
          href={product_grouping_helpers.fullPermalink(item.product_grouping, item.variant)}
          className={styles.cmCartDropdown_Item_NameImageWrapper}>
          <div className={styles.cmCartDropdown_Item_SecondaryContent}>
            <img
              className={styles.cmCartDropdown_Item_Image}
              src={product_grouping_helpers.getThumb(item.product_grouping, item.variant)}
              alt={`${item.product_grouping.name} - ${item.variant.volume}`} />
          </div>
          <div
            href={product_grouping_helpers.fullPermalink(item.product_grouping, item.variant)}
            className={cn('cmCartDropdown_Item_PrimaryContent', 'cmCartDropdown_Item_Details')}>
            <MBText.Span className={styles.cmCartDropdown_ItemName}>{item.product_grouping.name}</MBText.Span>
            <MBText.Span className={styles.cmCartDropdown_ItemVolume}>
              {item.variant.volume} {I18n.t('ui.nav.cart_dropdown.item_quantity', {count: item.quantity})}
            </MBText.Span>
            <MBText.Span className={styles.cmCartDropdown_ItemUnitPrice}>{formatCurrency(item.variant.price)}</MBText.Span>
          </div>
        </MBLink.View>
        <div className={cn('cmCartDropdown_Item_SecondaryContent', 'cmCartDropdown_Item_Price')}>
          <MBText.Span>{formatCurrency(cart_item_helpers.itemSubtotal(item))}</MBText.Span>
        </div>
        {show_remove_button && (
          <div className={cn('cmCartDropdown_Item_SecondaryContent')}>
            <RemoveItemLink cart_item_id={item.id} removeCartItem={removeCartItem} />
          </div>
        )}
      </div>
    </li>
  );
};

type OutOfStockCellProps = {
  id: number,
  name: string,
  price: string,
  quantity: number,
  total: string,
  volume: string
}

const OutOfStockCell = ({item}: OutOfStockCellProps) => {
  if (!item) return null;

  return (
    <li className={styles.cmCartDropdown_ItemRow_Animation}>
      <div className={styles.cmCartDropdown_ItemRow}>
        <div className={styles.cmCartDropdown_Item_SecondaryContent}>
          <img
            className={styles.cmCartDropdown_Out_Of_Stock_Image}
            src={'/assets/components/compounds/product_scroller/bottle_outline.png'}
            srcSet={'/assets/components/compounds/product_scroller/bottle_outline@2x.png 2x, ' +
              '/assets/components/compounds/product_scroller/bottle_outline@3x.png 3x'}
            alt={`${item.name} - ${item.volume}`} />
        </div>
        <div className={cn('cmCartDropdown_Item_PrimaryContent', 'cmCartDropdown_Item_Details')}>
          <MBText.Span className={styles.cmCartDropdown_ItemName}>{item.name}</MBText.Span>
          <MBText.Span className={styles.cmCartDropdown_ItemOutOfStock}>
            This product is currently out of stock
          </MBText.Span>
          <MBText.Span className={styles.cmCartDropdown_ItemVolume}>
            {item.volume} {I18n.t('ui.nav.cart_dropdown.item_quantity', {count: item.quantity})}
          </MBText.Span>
        </div>
        <div className={cn('cmCartDropdown_Item_SecondaryContent', 'cmCartDropdown_Item_Price')} />
      </div>
    </li>
  );
};

const RemoveItemLink = ({cart_item_id, removeCartItem}) => (
  <div
    role="button"
    tabIndex={0}
    className={styles.cmCartDropdown_Item_Remove}
    onClick={(e) => {
      e.stopPropagation();
      e.preventDefault();
      removeCartItem(cart_item_id);
      dispatchAction({
        actionType: 'track:cart:remove',
        variant_id: cart_item_id
      });
    }}>
    Ã—
  </div>
);

const ItemCellDTP = {removeCartItem: cart_item_actions.removeCartItem};
const ItemCellContainer = connect(null, ItemCellDTP)(ItemCell);
ItemCellContainer.OutOfStock = OutOfStockCell;

export default ItemCellContainer;
