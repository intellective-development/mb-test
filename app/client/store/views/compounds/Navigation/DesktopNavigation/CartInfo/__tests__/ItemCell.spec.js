import React from 'react';
import cart_item_factory from 'store/business/cart_item/__tests__/cart_item.factory';

import TestProvider from 'store/views/__tests__/utils/TestProvider';
import ItemCell from '../ItemCell';

describe('ItemCell', () => {
  it('renders', () => {
    expect(render(
      <TestProvider>
        <ItemCell item={cart_item_factory.build('with_product_grouping', 'five_dollar')} />
      </TestProvider>
    )).toMatchSnapshot();
  });

  it('renders with quantity > 1', () => {
    expect(render(
      <TestProvider>
        <ItemCell item={cart_item_factory.build('with_product_grouping', 'five_dollar', {quantity: 10})} />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
