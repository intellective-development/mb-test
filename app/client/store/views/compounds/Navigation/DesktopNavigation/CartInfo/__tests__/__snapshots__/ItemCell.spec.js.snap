// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCell renders 1`] = `
<li>
  <div>
    <a
      href="/store/product/budweiser-light-1/budweiser-light-6-pack-12oz-bottles"
    >
      <div>
        <img
          alt="Budweiser Light - 6 pack, 12oz bottles"
          src="https://placeholdit.imgix.net/~text?txtsize=33&txt=thumb&w=340&h=400"
        />
      </div>
      <div
        class="cmCartDropdown_Item_PrimaryContent cmCartDropdown_Item_Details"
        href="/store/product/budweiser-light-1/budweiser-light-6-pack-12oz-bottles"
      >
        <span
          class="elMBText elMBText_ResetSpacing"
        >
          Budweiser Light
        </span>
        <span
          class="elMBText elMBText_ResetSpacing"
        >
          6 pack, 12oz bottles 
        </span>
        <span
          class="elMBText elMBText_ResetSpacing"
        >
          $5.00
        </span>
      </div>
    </a>
    <div
      class="cmCartDropdown_Item_SecondaryContent cmCartDropdown_Item_Price"
    >
      <span
        class="elMBText elMBText_ResetSpacing"
      >
        $5.00
      </span>
    </div>
    <div
      class="cmCartDropdown_Item_SecondaryContent"
    >
      <div
        role="button"
        tabindex="0"
      >
        ×
      </div>
    </div>
  </div>
</li>
`;

exports[`ItemCell renders with quantity > 1 1`] = `
<li>
  <div>
    <a
      href="/store/product/budweiser-light-1/budweiser-light-6-pack-12oz-bottles"
    >
      <div>
        <img
          alt="Budweiser Light - 6 pack, 12oz bottles"
          src="https://placeholdit.imgix.net/~text?txtsize=33&txt=thumb&w=340&h=400"
        />
      </div>
      <div
        class="cmCartDropdown_Item_PrimaryContent cmCartDropdown_Item_Details"
        href="/store/product/budweiser-light-1/budweiser-light-6-pack-12oz-bottles"
      >
        <span
          class="elMBText elMBText_ResetSpacing"
        >
          Budweiser Light
        </span>
        <span
          class="elMBText elMBText_ResetSpacing"
        >
          6 pack, 12oz bottles (x10)
        </span>
        <span
          class="elMBText elMBText_ResetSpacing"
        >
          $5.00
        </span>
      </div>
    </a>
    <div
      class="cmCartDropdown_Item_SecondaryContent cmCartDropdown_Item_Price"
    >
      <span
        class="elMBText elMBText_ResetSpacing"
      >
        $50.00
      </span>
    </div>
    <div
      class="cmCartDropdown_Item_SecondaryContent"
    >
      <div
        role="button"
        tabindex="0"
      >
        ×
      </div>
    </div>
  </div>
</li>
`;
